dbschema/seeds/007.language-engagements.edgeql

```diff
@@ -0,0 +1,66 @@
+with
+  engagementsJson := to_json('[
+   {
+      "status": "InDevelopment",
+      "startDateOverride": "2020-04-01",
+      "endDateOverride": "2020-06-30",
+      "language": "English",
+      "lukePartnership": true,
+      "openToInvestorVisit": true,
+      "projectName": "Misty Mountains"
+    },
+    {
+      "status": "FinalizingCompletion",
+      "startDateOverride": "2016-04-01",
+      "endDateOverride": "2017-06-30",
+      "language": "Quenya",
+      "lukePartnership": true,
+      "openToInvestorVisit": false,
+      "projectName": "Arnor Lake"
+    },
+    {
+      "status": "Active",
+      "language": "Sindarin",
+      "lukePartnership": false,
+      "openToInvestorVisit": true,
+      "projectName": "Lothlorien"
+    },
+    {
+      "status": "Active",
+      "language": "Khuzdul",
+      "lukePartnership": false,
+      "openToInvestorVisit": true,
+      "projectName": "Emyn Muil"
+    },
+    {
+      "status": "FinalizingCompletion",
+      "language": "Westron",
+      "lukePartnership": false,
+      "openToInvestorVisit": false,
+      "paratextRegistryId": "1234567890",
+      "projectName": "South Downs"
+    }
+  ]'),
+  engagements := (
+    for engagement in json_array_unpack(engagementsJson)
+    union (
+      (select LanguageEngagement filter .language =
+        (select Language filter .displayName = <str>engagement['language'])) ??
+      (with
+        language := assert_single((select Language filter .name = <str>engagement['language'])),
+        project := (select Project filter .name = <str>engagement['projectName']),
+        insert LanguageEngagement {
```

Comment:
This with block could be moved up, and used in the select & insert.
```edgeql
with
  language := assert_single((select Language filter .name = <str>engagement['language'])),
  project := (select Project filter .name = <str>engagement['projectName'])
select (
  (select LanguageEngagement filter .language = language and .project = project) ??
  (insert LanguageEngagement {
    ...
  })
)
```

Reply:
I guess the way I had it wasn't as safe anyway, because I wasn't checking if the language/project combo existed before insert...just the language.